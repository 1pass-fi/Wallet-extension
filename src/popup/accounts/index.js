import React from 'react'
import './index.css'
import { Route, Switch } from 'react-router-dom'

import AccountHome from './accountHome'
import AccountImport from './accountImport'
import ImportPhrase from './importByPhrase/index'
import ImportFile from './importByFile/index'
import ImportFileSuccess from './importByFileSuccess/index'
import ImportPhraseSuccess from './importByPhraseSuccess/index'
import CreateWallet from './createWallet/index'
import Login from './accountLockScreen/index'

const Account = () => {
  return (
    <Switch>
      <Route exact path='/account'>
        <AccountHome />
      </Route>
      <Route path='/account/import/phrase/success'>
        <ImportPhraseSuccess />
      </Route>
      <Route path='/account/import/phrase'>
        <ImportPhrase />
      </Route>
      <Route path='/account/import/keyfile/success'>
        <ImportFileSuccess />
      </Route>
      <Route path='/account/import/keyfile'>
        <ImportFile />
      </Route>
      <Route path='/account/import'>
        <AccountImport />
      </Route>
      <Route path='/account/create'>
        <CreateWallet />
      </Route>
      <Route path='/account/login'>
        <Login />
      </Route>
    </Switch>
  )
}

export default Account
