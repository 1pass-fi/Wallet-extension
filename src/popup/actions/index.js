import { setActivities } from './activities'
import { setError } from './error'
import { importWallet } from './koi'

export default {
  setActivities,
  setError,
  importWallet
}
