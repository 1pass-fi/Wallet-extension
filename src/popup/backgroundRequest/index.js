import backgroundConnect from 'actions/backgroundConnect'

import { GalleryRequest } from './gallery'
import { AssetRequest } from './assets'
import { ActivityRequest } from './activities'
import { WalletRequest } from './wallet'

class BackgroundRequest {
  constructor(backgroundConnect) {
    this.backgroundConnect = backgroundConnect
    this.wallet = new WalletRequest(this.backgroundConnect)
    this.activities = new ActivityRequest(this.backgroundConnect)
    this.assets = new AssetRequest(this.backgroundConnect)
    this.gallery = new GalleryRequest(this.backgroundConnect)
  }
}

export const backgroundRequest = new BackgroundRequest(backgroundConnect)
