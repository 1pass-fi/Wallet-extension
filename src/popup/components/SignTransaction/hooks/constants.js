export const TRANSACTION_TYPE = {
  ORIGIN_TOKEN_TRANSFER: 'ORIGIN_TOKEN_TRANSFER',
  CUSTOM_TOKEN_TRANSFER: 'CUSTOM_TOKEN_TRANSFER',
  CONTRACT_DEPLOYMENT: 'CONTRACT_DEPLOYMENT',
  CONTRACT_INTERACTION: 'CONTRACT_INTERACTION'
}

export const TRANSACTION_METHOD = {
  SET_APPROVAL_FOR_ALL: 'SET_APPROVAL_FOR_ALL',
  MINT_COLLECTIBLES: 'MINT_COLLECTIBLES',
  APPROVE: 'APPROVE',
  CONTRACT_INTERACTION: 'CONTRACT_INTERACTION',
  CONTRACT_DEPLOYMENT: 'CONTRACT_DEPLOYMENT',
  TOKEN_TRANSFER: 'TOKEN_TRANSFER'
}

export const TAB = {
  DETAIL: 'DETAIL',
  DATA: 'DATA',
  SIMULATION: 'SIMULATION'
}
