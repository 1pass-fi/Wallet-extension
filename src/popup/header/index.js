import React from 'react'
import LogoIcon from 'img/koi-logo.svg'
import SettingIcon from 'img/settings-icon.svg'
import NavBar from './navBar'
import './index.css'

export default () => {
  return (
    <>
      <header>
        <LogoIcon className="logo" />
        <button className="gallery-button">My NFT Gallery</button>
        <button className="setting-button">
          <SettingIcon />
        </button>
      </header >
      <NavBar />
    </>
  )
}
