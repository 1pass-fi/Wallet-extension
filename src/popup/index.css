@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Sora', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

html::-webkit-scrollbar {
  display: none;
}

/* width */
::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

/* Track */
::-webkit-scrollbar-track {
  background: rgba(165, 165, 165, 0);
  border-radius: 10px;
  border: 0px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: #bef0ed;
  border-radius: 10px;
  border: 0px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: #555;
}
