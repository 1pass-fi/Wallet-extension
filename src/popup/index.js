/* istanbul ignore next */
import React from 'react'
import ReactDOM from 'react-dom'
import * as browser from 'webextension-polyfill'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'

import './index.css'
import Popup from './Popup'
import store from './store'

browser.runtime.sendMessage({ data: 'hello' })

ReactDOM.render(
  <Provider store={store}>
    <Router>
      <Popup />
    </Router>
  </Provider>,
  document.getElementById('root')
)
