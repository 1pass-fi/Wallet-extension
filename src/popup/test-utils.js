import React from 'react'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { render as rtlRender } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { applyMiddleware,createStore } from 'redux'
import thunk from 'redux-thunk'

import rootReducer from './reducers/index'

function render(
  ui,
  {
    history = createMemoryHistory(),
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export * from '@testing-library/react'
export { render }
