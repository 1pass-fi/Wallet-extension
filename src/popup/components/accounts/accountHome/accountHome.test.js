import React from 'react'
import { fireEvent } from '@testing-library/react'
import { TYPE } from 'constants/accountConstants'
import { createMemoryHistory } from 'history'
import { act, render, screen } from 'popup/test-utils'

import '@testing-library/jest-dom'

import AccountHome from '.'

describe('AccountHome', () => {
  const accounts = [
    {
      type: TYPE.ARWEAVE,
      address: 'fakeAddress',
      accountName: 'Test account',
    },
    {
      type: TYPE.ARWEAVE,
      address: 'fakeAddress',
      accountName: 'Test account',
    },
  ]

  let history
  beforeEach(() => {
    history = createMemoryHistory()
  })

  it('renders without crashing', () => {
    const { container } = render(<AccountHome />, {
      initialState: { accounts },
      history,
    })

    const globalBtn = container.querySelector('.global-button')

    expect(globalBtn).toBeInTheDocument()
    expect(() => screen.getByRole('form')).toThrow()
    expect(container.querySelectorAll('.wallet-wrapper')).toHaveLength(
      accounts.length
    )
  })

  describe('User clicks global send button', () => {
    describe('Send Koi form is not opening', () => {
      it('opens send form', async () => {
        const { container } = render(<AccountHome />, { history })

        const globalBtn = container.querySelector('.global-button')

        await act(async () => {
          await fireEvent.click(globalBtn)
        })

        expect(history.location.search).toBe('?openSendForm=true')
        expect(container.querySelector('form')).toBeInTheDocument()
      })
    })

    describe('Send Koi form is opening', () => {
      it('removes the search params', async () => {
        const { container } = render(<AccountHome />, { history })

        const globalBtn = container.querySelector('.global-button')

        await act(async () => {
          await fireEvent.click(globalBtn)
          await fireEvent.click(globalBtn)
        })

        expect(history.location).toEqual(
          expect.objectContaining({
            pathname: '/account',
            search: '',
          })
        )
      })
    })
  })

  describe('User clicks to add account', () => {
    it('redirects to welcome', async () => {
      const { container } = render(<AccountHome />, { history })

      const addAccountBtn = container.querySelector('.button')

      await act(async () => {
        await fireEvent.click(addAccountBtn)
      })

      expect(history.location.pathname).toBe('/account/welcome')
    })
  })
})
