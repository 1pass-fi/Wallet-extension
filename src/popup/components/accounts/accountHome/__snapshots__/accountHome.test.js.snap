// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for AccountHome component Render without crashing renders correctly 1`] = `
<div>
  <div>
    <a
      class="plus-button"
      href="/account/import"
    >
      <svg />
    </a>
  </div>
</div>
`;
